block/base/tx_priority_test.go

```diff
@@ -0,0 +1,223 @@
+package base_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/skip-mev/block-sdk/block/base"
+	"github.com/skip-mev/block-sdk/testutils"
+)
+
+const maxUint64 = "18446744073709551616" // value is 2^64
+
+func TestTxPriority(t *testing.T) {
+	acct := testutils.RandomAccounts(rand.New(rand.NewSource(1)), 1)
+	txc := testutils.CreateTestEncodingConfig().TxConfig
+
+	type testCase struct {
+		name     string
+		priority base.TxPriority[string]
+	}
+
+	testCases := []testCase{
+		{
+			"DeprecatedTxPriority",
+			base.DeprecatedTxPriority(),
+		},
+		{
+			"DefaultTxPriority",
+			base.DefaultTxPriority(),
+		},
+	}
+	t.Run("test getting a tx priority: DefaultTxPriority", func(t *testing.T) {
+		priority := base.DefaultTxPriority()
+
+		tx, err := testutils.CreateTx(txc, acct[0], 0, 0, nil, sdk.NewCoin("stake", math.NewInt(1)))
+		require.NoError(t, err)
+
+		require.Equal(t, "1,stake", priority.GetTxPriority(nil, tx))
+	})
+
+	t.Run("test with amt that is not uint64", func(t *testing.T) {
```

Comment:
nit: This test name is confusing. It's actually just testing two txs with a single equivalent denom and one value strictly greater than the other.

Reply:
This is testing that non-uint64 coin values work as expected (i.e `newIntFromString`), we need to do a comparison operation as that is where the value is ultimately read from a string. Happy to export + test the method specifically, but I felt this was simpler?

Reply:
I see, makes sense.